tests/zfs-tests/cmd/read_dos_attributes/read_dos_attributes.c

```diff
@@ -0,0 +1,51 @@
+/*
+ * This file and its contents are supplied under the terms of the
+ * Common Development and Distribution License ("CDDL"), version 1.0.
+ * You may only use this file in accordance with the terms of version
+ * 1.0 of the CDDL.
+ *
+ * A full copy of the text of the CDDL should have accompanied this
+ * source.  A copy of the CDDL is also available via the Internet at
+ * http://www.illumos.org/license/CDDL.
+ */
+
+/*
+ * Copyright 2022 iXsystems, Inc.
+ */
+
+/*
+ * FreeBSD allows to update and retreive additional file level attributes.
+ * For Linux, two IOCTLs have been added to update and retrieve additional
+ * level attributes.
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <stdint.h>
+#include <fcntl.h>
+#include <unistd.h>
+#include <sys/ioctl.h>
+#include <err.h>
+
+#define	ZFS_IOC_GETDOSFLAGS	_IOR(0x83, 1, uint64_t)
+
+int
+main(int argc, const char * const argv[])
+{
+	if (argc != 2)
+		errx(EXIT_FAILURE, "Usage: %s filename", argv[0]);
+
+	int fd = open(argv[1], O_RDONLY);
+	if (fd < 0)
+		err(EXIT_FAILURE, "Failed to open %s", argv[1]);
+
+	uint64_t dosflags = -1;
+	if (ioctl(fd, ZFS_IOC_GETDOSFLAGS, &dosflags) == -1)
+		err(EXIT_FAILURE, "ZFS_IOC_GETDOSFLAGS failed");
+
+	(void) close(fd);
+
+	(void) printf("Dos Flags: 0x%lx", dosflags);
```

Comment:
IIRC this is how we have been printing 64-bit values:
```suggestion
	(void) printf("Dos Flags: 0x%llx", (unsigned long long)dosflags);
```

Reply:
```c
(void) printf("Dos Flags: 0x%" PRIx64 "", dosflags);
```
works too, esp. since this is stand-alone; unrelatedly, it'd be even better if it were
```c
(void) printf("Dos Flags: %#" PRIx64 "", dosflags);
```

Reply:
The one I commonly see in the codebase is `printf("%llx", (u_longlong_t)foo);`

```
zfs $ git grep '(u_longlong_t)' | wc -l
1020
zfs $ git grep '(unsigned long long)' | wc -l
45
zfs $ git grep 'PRIx64' | wc -l
15
```

Whether this is a bug or a feature is a different question.

Reply:
We've tried to use `printf("%llx", (u_longlong_t)foo);` is most cases, it's not ideal but in practice we've found it's the least likely to generate a warning on all architectures, compilers, and kernel versions.